probe/kubernetes/controls.go

```diff
@@ -78,33 +78,21 @@ func (r *Reporter) CapturePod(f func(xfer.Request, string, string) xfer.Response
 func (r *Reporter) CaptureResource(f func(xfer.Request, string, string, string) xfer.Response) func(xfer.Request) xfer.Response {
 	return func(req xfer.Request) xfer.Response {
 		var resource, uid string
-		for _, parser := range []struct {
-			res string
-			f   func(string) (string, bool)
-		}{
-			{report.Deployment, report.ParseDeploymentNodeID},
-		} {
-			if u, ok := parser.f(req.NodeID); ok {
-				resource, uid = parser.res, u
-				break
-			}
+		if u, ok := report.ParseDeploymentNodeID(req.NodeID); ok {
+			resource, uid = report.Deployment, u
 		}
 		if resource == "" {
 			return xfer.ResponseErrorf("Invalid ID: %s", req.NodeID)
 		}
-
-		switch resource {
-		case report.Deployment:
-			var deployment Deployment
-			r.client.WalkDeployments(func(d Deployment) error {
-				if d.UID() == uid {
-					deployment = d
-				}
-				return nil
-			})
-			if deployment != nil {
-				return f(req, "deployment", deployment.Namespace(), deployment.Name())
+		var deployment Deployment
+		r.client.WalkDeployments(func(d Deployment) error {
+			if d.UID() == uid {
+				deployment = d
 			}
+			return nil
+		})
+		if deployment != nil {
+			return f(req, "deployment", deployment.Namespace(), deployment.Name())
```

Comment:
I suggest making the structure the same here as in `CapturePod`, which a) has the condition reversed, and b) has `f` take just three arguments. The latter will require removing `resource` from the signature of `Reporter.Scale{Up,Down}`, and instead calling the client library with "deployment" as a constant. Incidentally, this will actually make `Reporter.Scale{Up,Down}` do what their doc comment says: "control to scale up a _deployment_".

Reply:
The change I proposed to `Reporter.Scale{Up,Down}` is still outstanding.